aea/runtime.py

```diff
@@ -125,6 +135,17 @@ def __init__(
         self._async_stop_lock: Optional[asyncio.Lock] = None
         self._task: Optional[asyncio.Task] = None
 
+    def set_loop(self, loop: AbstractEventLoop) -> None:
+        """
+        Set event loop to be used.
+
+        :param loop: event loop to use.
+        """
+        super().set_loop(loop)
+        self._agent._multiplexer.set_loop(self._loop)
```

Comment:
@solarw somewhat unrelated to this PR; but your use of private and public methods/attributes is not consistent. Here It should be `self._agent.multiplexer.set_loop(self._loop)`. Maybe, both `_loop` and `_agent` could be exposed as a property; but that depends on whether we want to expose them as public.

Reply:
yes, it's not the best way, assume to update on refactoring, when will have a clue about all dependencies.